Content.Tests/DMProject/Tests/Builtins/__IMPLIED_TYPE__.dm

```diff
@@ -4,4 +4,8 @@
 /proc/RunTest()
 	var/datum/test/D = __IMPLIED_TYPE__
 	ASSERT(D.bar == "foobar")
+	D = ArgumentTest(__IMPLIED_TYPE__)
+	ASSERT(D.bar == "foobar")
```

Comment:
This second assert is redundant

Reply:
it's not redundant, we reassign D. we *must* reassign D or else there is no implied type for the arguments, and since we reassign D, redoing the assert is the easiest way i could think of to check it
that said i think it would probably be better both times to just check `ASSERT(D == /datum/test)`